arrows/core/metadata_map_io_csv.cxx

```diff
@@ -24,6 +31,49 @@ namespace arrows {
 
 namespace core {
 
+namespace {
+
+// ----------------------------------------------------------------------------
+template < typename out >
+void
+split( std::string const& s, char delim, out result )
+{
+  std::istringstream iss( s );
+  std::string item;
+
+  while( std::getline( iss, item, delim ) )
+  {
+    *result++ = item;
+  }
+}
+
+// ----------------------------------------------------------------------------
+std::vector< std::string >
+split( std::string const& s, char delim )
+{
+  std::vector< std::string > elems;
+  split( s, delim, std::back_inserter( elems ) );
+  return elems;
+}
+
+// ----------------------------------------------------------------------------
+std::string
+trim( std::string const& str, std::string const& whitespace = " \t" )
+{
+  auto const str_begin = str.find_first_not_of( whitespace );
+
+  if( str_begin == std::string::npos )
+  {
+    return ""; // no content
+  }
+  auto const str_end = str.find_last_not_of( whitespace );
+  auto const str_range = str_end - str_begin + 1;
+
+  return str.substr( str_begin, str_range );
+}
+
+} // namespace
```

Comment:
I believe this was introduced by uncrustify, so maybe we should be consistent with that. Not 100% sure though.

Reply:
Probably, but I haven't yet taught it to make the comment `// namespace <anonymous>` 🙂. Anyway, since it currently leaves existing comments alone (also on my to-do list to "fix"), changing it won't result in uncrustify wanting to change it back.